lib/cryptoexchange/exchanges/basefex/services/contract_stat.rb

```diff
@@ -0,0 +1,51 @@
+module Cryptoexchange::Exchanges
+  module Basefex
+    module Services
+      class ContractStat < Cryptoexchange::Services::Market
+        class << self
+          def supports_individual_ticker_query?
+            false
+          end
+        end
+
+        def fetch
+          output = super ticker_url
+          adapt_all(output)
+        end
+
+        def ticker_url
+          "#{Cryptoexchange::Exchanges::Basefex::Market::API_URL}/instruments"
+        end
+
+        def adapt_all(output)
+          output.map do |pair|
+            base, target = Basefex::Market.separate_symbol(pair["symbol"])
+            market_pair = Cryptoexchange::Models::MarketPair.new(
+                            base: base,
+                            target: target,
+                            inst_id: pair["symbol"],
+                            contract_interval: "perpetual",
+                            market: Basefex::Market::NAME
+                          )
+
+            adapt(pair, market_pair)
+          end
+        end
+
+        def adapt(output, market_pair)
+          contract_stat = Cryptoexchange::Models::ContractStat.new
+          contract_stat.base = market_pair.base
+          contract_stat.target = market_pair.target
+          contract_stat.market = Basefex::Market::NAME
+          contract_stat.contract_type = "perpetual"
+          contract_stat.open_interest = output['openValue']
+          contract_stat.funding_rate_percentage = output["fundingRate"]
```

Comment:
fundingRate need to multiply by 100? verify here https://www.basefex.com/trade/BTCUSD
make sure the funding rate is tally